http-client/src/main/java/io/airlift/http/client/jetty/JettyResponse.java

```diff
@@ -31,6 +32,16 @@ public int getStatusCode()
         return response.getStatus();
     }
 
+    @Override
+    public HttpVersion getHttpVersion()
+    {
+        return switch (response.getVersion()) {
+            case HTTP_1_1 -> HttpVersion.HTTP_1_1;
+            case HTTP_2 -> HttpVersion.HTTP_2;
+            default -> throw new IllegalStateException("Unexpected HTTP version: " + response.getVersion());
```

Comment:
This will break if the server uses `HTTP_3`. I think we need to add an `HTTP_3` enum, or make this return optional.

Reply:
Right, fixed, now the switch is exhaustive